src/commands/deploy/utils.ts

```diff
@@ -201,17 +198,37 @@ function getNoTargetPackage(errorMessage: string): string | null {
 
 export async function addSSRComponentToConfig(
     configPath: string,
-    config: YamlProjectConfiguration | RawYamlProjectConfiguration,
     component: SSRFrameworkComponent,
     componentType: SSRFrameworkComponentType,
 ) {
     const configIOController = new YamlConfigurationIOController(configPath);
+    // We have to read the config here with fillDefaults=false
+    // to be able to edit it in the least intrusive way
+    const config = await configIOController.read(/* fillDefaults= */ false);
+
     const relativePath = path.relative(process.cwd(), component.path) || ".";
 
+    // Ensure each script field is an array
+    // It's easier to use arrays consistently instead of
+    // having to check if it's a string or an array
+    const scripts = component.scripts;
+    if (scripts) {
+        if (scripts.deploy && typeof scripts.deploy === "string") {
+            scripts.deploy = [scripts.deploy];
+        }
+        if (scripts.build && typeof scripts.build === "string") {
+            scripts.build = [scripts.build];
+        }
+        if (scripts.start && typeof scripts.start === "string") {
+            scripts.start = [scripts.start];
+        }
+    }
```

Comment:
There’s a duplicated code segment here, matching lines 10-24 in `analyze/utils.ts`. If this logic is likely to stay consistent, consider extracting it into a reusable function to improve maintainability.

Reply:
Done ✅